File: archives/C#/0242. Valid Anagram.cs
public class Solution {
    public bool IsAnagram(string s, string t) {
        if(s.Length!=t.Length){
            return false;
        }
        IDictionary<char,int> cmpDict=new Dictionary<char,int>();
        foreach(char ss in s){
            if(!cmpDict.ContainsKey(ss)){
                cmpDict[ss]=1;
            }
            else{
                cmpDict[ss]+=1;
            }
        }
        bool rep=true;
        foreach(char ss in t){
            if(cmpDict.ContainsKey(ss)){
                cmpDict[ss]-=1;
            }
            else{
                rep=false;
                break;
            }
        }
        
        if(rep==false){return false;}
        foreach(char key in cmpDict.Keys){
            if(cmpDict[key]!=0){
                rep=false;
                break;
            }
        }
        return rep;
    }
}

File: archives/20190519python/0242. Valid Anagram.py
class Solution:
    def isAnagram(self, s: str, t: str) -> bool:
        if len(s)!=len(t):
            return False
        dict_st={}
        for ss in s:
            if ss in dict_st:
                dict_st[ss]+=1
            else:
                dict_st[ss]=1
        
        for tt in t:
            if tt in dict_st and dict_st[tt]>=1:
                dict_st[tt]-=1
            else:
                return False
        
        return True
